pkg/kubernetes/pvc/v1alpha1/pvc.go

```diff
@@ -0,0 +1,69 @@
+package v1alpha1
+
+import (
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+type pvcFiltersFn func(pvc) bool
+
+type pvc struct {
+	// actual pvc object
+	object v1.PersistentVolumeClaim
+}
+
+type pvcList struct {
+	// list of pvc objects
+	items []pvc
+}
+
+type listBuilder struct {
+	list *pvcList
+}
+
+// ListBuilder enables building an instance
+// of pvcList
+func ListBuilder() *listBuilder {
+	return &listBuilder{list: &pvcList{}}
+}
+
+// HasName is filter function to filter pvc's
+// based on the name
+func HasName(name string) pvcFiltersFn {
+	return func(obj pvc) bool {
+		return strings.Contains(obj.object.GetName(), name)
+	}
+}
+
+// WithListObject builds the list of pvc instances
+// based on the provided pvc instances
+func (b *listBuilder) WithListObject(list *v1.PersistentVolumeClaimList) *listBuilder {
+	if list == nil {
+		return b
+	}
+	for _, c := range list.Items {
+		b.list.items = append(b.list.items, pvc{c})
+	}
+	return b
+}
+
+// AddCheck adds filters to filter the pvc list
+func (b *listBuilder) AddCheck(checks ...pvcFiltersFn) *listBuilder {
+	for _, o := range checks {
+		p := []pvc{}
+		for _, i := range b.list.items {
+			if o(i) {
```

Comment:
The purpose of this function is to add filter(s) i.e. predicates. The purpose of this function is never to execute these filters. These filter predicates will be executed during `List()` call.
Hence, modify the listBuilder as below:
```go
type listBuilder struct {
  list       *pvcList
  filters   []predicate
}
```

Reply:
done